### PR TITLE
Update the CI build to only run p0 UI tests for PRs and all UI tests for rolling CI builds

### LINKED ISSUE #933: Update builds to only run highest priority UI tests only.
**Describe the bug**

Currently all the UI tests are run as part of the CI builds. With new UI tests coming online, only the highest priority and most reliable tests should be run as part of the CI builds. The remaining tests can be run locally and during monthly test passes.

### PR DESCRIPTION
**Pull request by sanderl**
### Description of the changes:
- Updated the CI build yaml to conditionally run UI tests that are not P0
- Added a new UI tests template for the PR UI tests.

### How changes were validated:

- Tested using a private pipeline to ensure the right tests were being picked up through private PR and rolling build.
-
-

### CODE REVIEW

**Comment by sanderl** [comment]
> I really don't like the idea of forking run-ui-tests.yaml.
> 
> I found out (via [this issue](https://github.com/MicrosoftDocs/vsts-docs/issues/6989)) that you can accomplish this more simply by using conditional insertion syntax. The change would look something like this: [mcooley@125f36a](https://github.com/mcooley/calculator/commit/125f36aedb1d917ddb56f08826d3888dc6ebe7aa)
> 
> Let's do that instead.

I just updated my PR to use your change. I'm testing it out now. I should have an update for the testing soon.

**Comment by sanderl** [comment]
> > I really don't like the idea of forking run-ui-tests.yaml.
> > I found out (via [this issue](https://github.com/MicrosoftDocs/vsts-docs/issues/6989)) that you can accomplish this more simply by using conditional insertion syntax. The change would look something like this: [mcooley@125f36a](https://github.com/mcooley/calculator/commit/125f36aedb1d917ddb56f08826d3888dc6ebe7aa)
> > Let's do that instead.
> 
> I just updated my PR to use your change. I'm testing it out now. I should have an update for the testing soon.

Both tests passed. The PR build only ran P0 and the full CI build ran all of the UI tests.

File: build/pipelines/templates/run-ui-tests.yaml [hunk 1]
```diff
@@ -8,7 +8,6 @@ jobs:
 - job: UITests${{ parameters.platform }}
   displayName: UITests ${{ parameters.platform }}
   dependsOn: Build${{ parameters.platform }}
-  condition: succeeded()
```
Comment (mcooley) [inline build/pipelines/templates/run-ui-tests.yaml]: We still need condition: succeeded(), right?

File: build/pipelines/templates/run-ui-tests.yaml [hunk 2]
```diff
@@ -39,12 +38,14 @@ jobs:
     inputs:
       filePath: $(Build.ArtifactStagingDirectory)/drop/Release/${{ parameters.platform }}/Calculator/AppPackages/Calculator_$(Build.BuildNumber)_Test/Add-AppDevPackage.ps1
       arguments: -Force
-
+
```
Comment (mcooley) [inline build/pipelines/templates/run-ui-tests.yaml]: would be nice to remove this trailing whitespace too while you're at it

**Pull request merged**